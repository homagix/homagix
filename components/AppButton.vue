<script setup lang="ts">
const emit = defineEmits(["click"])
</script>

<template>
  <a @click.prevent="() => emit('click')">
    <slot />
  </a>
</template>

<style lang="scss" scoped>
a {
  display: inline-block;
  font-size: 150%;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  text-align: center;

  &:hover {
    background: #eeeeee;
  }

  &:active {
    background: #dddddd;
  }

  &.small {
    font-size: 115%;
    line-height: 0.9;
    width: 24px;
    height: 24px;
  }
}
</style>
