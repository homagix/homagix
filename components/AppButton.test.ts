import { describe, it, expect } from "vitest"
import { setup } from "@nuxt/test-utils"
import { mountSuspended } from "@nuxt/test-utils/runtime"
import AppButton from "./AppButton.vue"

describe("AppButton", () => {
  setup({ server: false })

  it("renders as expected", async () => {
    const wrapper = await mountSuspended(AppButton, { slots: { default: () => "test-button" }})
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("emits click event when anchor tag is clicked", async () => {
    const wrapper = await mountSuspended(AppButton)
    expect(wrapper.emitted()).toEqual({})
    await wrapper.find("a").trigger("click")
    expect(wrapper.emitted("click")).toHaveLength(1)
  })
})
