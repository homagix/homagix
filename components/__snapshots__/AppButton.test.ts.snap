// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppButton > renders as expected 1`] = `"<a data-v-2f131dae="">test-button</a>"`;
